src/ServiceControl.Monitoring/Metrics/Raw/LongValueOccurrences.cs

```diff
@@ -1,23 +1,63 @@
 ﻿namespace ServiceControl.Monitoring.Metrics.Raw
 {
+    using System;
+    using System.Collections.Concurrent;
     using NServiceBus;
 
     public class LongValueOccurrences : IMessage
     {
-        public long Version;
+        public class Pool
+        {
+            public static readonly Pool Default = new Pool();
+            ConcurrentStack<LongValueOccurrences> pool = new ConcurrentStack<LongValueOccurrences>();
 
-        public long BaseTicks;
+            public LongValueOccurrences Lease()
+            {
+                LongValueOccurrences value;
+                if (pool.TryPop(out value))
+                {
+                    return value;
+                }
 
-        public int[] Ticks;
+                return new LongValueOccurrences();
+            }
 
-        public long[] Values;
-    }
+            public void Release(LongValueOccurrences message)
+            {
+                message.index = InitialIndex;
+                Array.Clear(message.entries, 0, MaxEntries);
+                pool.Push(message);
+            }
+        }
```

Comment:
The pool size is effectively limited by the concurrency limit of the transport multiplied by the number of `LongValueOccurences` messages that a single message can be split into. I cannot see any reason to put another limitation on top of it. Is there any?